main.go

```diff
@@ -91,61 +110,72 @@ func main() {
 		os.Exit(1)
 	}
 
 	filterDatasetControllerURL, err := url.Parse(config.FilterDatasetControllerURL)
 	if err != nil {
-		log.Event(nil, "configuration value is invalid", log.Data{"config_name": "FilterDatasetControllerURL", "value": config.FilterDatasetControllerURL}, log.Error(err))
+		log.Event(ctx, "configuration value is invalid", log.Data{"config_name": "FilterDatasetControllerURL", "value": config.FilterDatasetControllerURL}, log.Error(err))
 		os.Exit(1)
 	}
 
 	geographyControllerURL, err := url.Parse(config.GeographyControllerURL)
 	if err != nil {
-		log.Event(nil, "configuration value is invalid", log.Data{"config_name": "GeographyControllerURL", "value": config.GeographyControllerURL}, log.Error(err))
+		log.Event(ctx, "configuration value is invalid", log.Data{"config_name": "GeographyControllerURL", "value": config.GeographyControllerURL}, log.Error(err))
 		os.Exit(1)
 	}
 
 	redirects.Init(assets.Asset)
 
 	router := pat.New()
 
-	router.Path("/healthcheck").HandlerFunc(hc.Do)
-
 	middleware := []alice.Constructor{
 		requestID.Handler(16),
 		log.Middleware,
 		securityHandler,
 		redirects.Handler,
 	}
 
+	zebedeeClient := client.New(config.ZebedeeURL)
+
 	if config.DatasetRoutesEnabled == true {
 		middleware = append(middleware, allRoutes.Handler(map[string]http.Handler{
 			"dataset_landing_page": reverseProxy.Create(datasetControllerURL, nil),
-		}))
+		}, zebedeeClient))
 	}
 
 	alice := alice.New(middleware...).Then(router)
 
 	babbageURL, err := url.Parse(config.BabbageURL)
 	if err != nil {
-		log.Event(nil, "configuration value is invalid", log.Data{"config_name": "BabbageURL", "value": config.BabbageURL}, log.Error(err))
+		log.Event(ctx, "configuration value is invalid", log.Data{"config_name": "BabbageURL", "value": config.BabbageURL}, log.Error(err))
 		os.Exit(1)
 	}
 
 	downloaderURL, err := url.Parse(config.DownloaderURL)
 	if err != nil {
-		log.Event(nil, "configuration value is invalid", log.Data{"config_name": "DownloaderURL", "value": config.DownloaderURL}, log.Error(err))
+		log.Event(ctx, "configuration value is invalid", log.Data{"config_name": "DownloaderURL", "value": config.DownloaderURL}, log.Error(err))
 		os.Exit(1)
 	}
 
 	searchHandler, err := analytics.NewSearchHandler()
 	if err != nil {
-		log.Event(nil, "error creating search analytics handler", log.Error(err))
+		log.Event(ctx, "error creating search analytics handler", log.Error(err))
 		os.Exit(1)
 	}
 
+	// Healthcheck API
+	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
+	if err != nil {
+		log.Event(ctx, "initialising healthcheck without versionInfo", log.Error(err))
+	}
+	hc := healthcheck.New(versionInfo, config.HealthckeckCriticalTimeout, config.HealthckeckInterval)
+	if err = hc.AddCheck("Zebedee", zebedeeClient.Checker); err != nil {
+		log.Event(ctx, "Failed to add Zebedee checker to healthcheck", log.Error(err))
```

Comment:
Again I would force app to close due to error; we wouldn't want the application to keep running without the ability to inform us of something being wrong